File: 0075-sort-colors/0075-sort-colors.py
class Solution(object):
    def sortColors(self, nums):
        """
        :type nums: List[int]
        :rtype: None Do not return anything, modify nums in-place instead.
        """
        low = 0
        mid = 0
        high = len(nums) - 1
        
        while(mid <= high):
            if(nums[mid] == 0):
                temp = nums[low]
                nums[low] = nums[mid]
                nums[mid] = temp
                mid+=1
                low+=1
            elif(nums[mid] == 2):
                temp = nums[high]
                nums[high] = nums[mid]
                nums[mid] = temp
                high-=1
            elif(nums[mid] == 1):
                mid+=1
            
        return nums

File: 0075-sort-colors/0075-sort-colors.js
/**
 * @param {number[]} nums
 * @return {void} Do not return anything, modify nums in-place instead.
 */
var sortColors = function(nums) {
//     Better - > get counts of all colors and modify given array
    let c_r = 0
    let c_b = 0
    let c_w = 0
    for (let i = 0; i< nums.length; i++) {
        if (nums[i] == 0) {
            c_r++
        }
        if (nums[i] == 1) {
            c_w++
        }
        else {
            c_b++
        }
    }
    for (let i = 0; i< nums.length; i++) {
        if (c_r) {
            nums[i] = 0
            c_r --
            continue
        }
        if (c_w) {
            nums[i] = 1
            c_w--
            continue
        }
        if (c_b) {
            nums[i] = 2
            c_b--
            continue
        }
    }
};